core/config/config_test.go

```diff
@@ -1,10 +1,8 @@
-package config_test
+package config
```

Comment:
according to golang docs, since this file ends in `_test.go` it won't be included in our final binary even if the `config` package is linked. Therefore, I've dropped the _test package suffix here, to allow us to drop the public exports of the readBackendConfig functions

Reply:
I guess this is more a bit of a code/style in writing tests. I always prefer (especially in go) to go with black box testing rather then white box testing - in the long run black box tests are less brittle and allow me to test the intention rather then corner cases, but I do agree that here things are quite shady and can be improved